File: schemas/booking-scheduling/drizzle/booking_attendees.ts
// booking_attendees: individual participants attached to a booking.
// See README.md for full design rationale.

import { pgTable, pgEnum, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { bookings } from "./bookings";
import { users } from "./users";

export const attendeeStatusEnum = pgEnum("attendee_status", ["confirmed", "cancelled", "no_show"]);

export const bookingAttendees = pgTable(
  "booking_attendees",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    bookingId: uuid("booking_id").notNull().references(() => bookings.id, { onDelete: "cascade" }),
    userId: uuid("user_id").references(() => users.id, { onDelete: "set null" }),
    name: text("name").notNull(),
    email: text("email").notNull(),
    phone: text("phone"),
    status: attendeeStatusEnum("status").notNull().default("confirmed"),
    cancelledAt: timestamp("cancelled_at", { withTimezone: true }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_booking_attendees_booking_id").on(table.bookingId),
    index("idx_booking_attendees_user_id").on(table.userId),
    index("idx_booking_attendees_email").on(table.email),
  ]
);

File: schemas/booking-scheduling/spacetimedb/booking_attendees.rs
// booking_attendees: Individual attendees associated with a booking.
// See README.md for full design rationale.

use spacetimedb::{SpacetimeType, Timestamp};

#[derive(SpacetimeType, Clone)]
pub enum AttendeeStatus {
    Confirmed, // type: String
    Cancelled,
    NoShow,
}

#[spacetimedb::table(name = booking_attendees, public)]
pub struct BookingAttendee {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub booking_id: String, // UUID — FK → bookings.id (cascade delete)
    #[index(btree)]
    pub user_id: Option<String>, // UUID — FK → users.id (set null)
    pub name: String,
    #[index(btree)]
    pub email: String,
    pub phone: Option<String>,
    pub status: AttendeeStatus,
    pub cancelled_at: Option<Timestamp>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
